src/main/java/org/broadinstitute/hellbender/engine/datasources/ReferenceAPISource.java

```diff
@@ -76,10 +80,50 @@ public ReferenceAPISource( final PipelineOptions pipelineOptions, final String r
         // If we go with Spark, we'll end up refactoring this to store all of the secrets
         // directly in this class and get rid of PipelineOptions as an argument to
         // getReferenceBases.
+        this.apiKey = pipelineOptions.as(GCSOptions.class).getApiKey();
+    }
+
+    private ReferenceAPISource(final PipelineOptions pipelineOptions, final Map<String, Reference> referenceMap){
         Utils.nonNull(pipelineOptions);
+        this.referenceMap = referenceMap;
+        this.referenceNameToIdTable = getReferenceNameToIdTableFromMap(referenceMap);
+
+        // For Spark, we keep around the apiKey from the PipelineOptions since we don't have
+        // a "context" with PipelineOptions available on each worker.
+        // If we go with Spark, we'll end up refactoring this to store all of the secrets
+        // directly in this class and get rid of PipelineOptions as an argument to
+        // getReferenceBases.
         this.apiKey = pipelineOptions.as(GCSOptions.class).getApiKey();
     }
 
+    /**
+     * Creates this ReferenceAPISource from an assembly ID by querying in the cloud APIs.
+     */
+    public static ReferenceAPISource fromReferenceSetAssemblyID(final PipelineOptions pipelineOptions, final String referenceSetAssemblyID){
+        Utils.nonNull(pipelineOptions);
+        Utils.nonNull(referenceSetAssemblyID);
+        final SearchReferenceSetsRequest content = new SearchReferenceSetsRequest();
+        content.setAssemblyId(referenceSetAssemblyID);
+        try {
+            final Genomics genomicsService = createGenomicsService(pipelineOptions);
+            final SearchReferenceSetsResponse found = genomicsService.referencesets().search(content).execute();
+            final Set<String> referenceSetIds = found.getReferenceSets().stream().map(rs -> rs.getId()).collect(Collectors.toSet());
+            final Map<String, Reference> ret = new HashMap<>();
+            for (final String rId : referenceSetIds) {
```

Comment:
Should we instead verify that there is only one and if not, throw an error (ideally with the two options that can be specified directly)? That's be a good test for the regular + decoy problem we have.